lib/puppet/indirector/rest.rb

```diff
@@ -53,22 +53,26 @@ def self.srv_service
   # ELSE
   #    Go for the legacy server/masterport settings, and hope for the best
   def self.server
-    setting = server_setting()
-    if setting && setting != :server && Puppet.settings.set_by_config?(setting)
-      Puppet.settings[setting]
-    else
-      server = Puppet.lookup(:server) do
-        if primary_server = Puppet.settings[:server_list][0]
-          Puppet.debug "Dynamically-bound server lookup failed; using first entry"
-          primary_server[0]
-        else
-          setting ||= :server
-          Puppet.debug "Dynamically-bound server lookup failed, falling back to #{setting} setting"
-          Puppet.settings[setting]
+    unless @server
+      setting = server_setting()
+      if setting && setting != :server && Puppet.settings.set_by_config?(setting)
+        Puppet.debug _("Selected server from the %{setting} setting: %{server}") % {setting: setting, server: Puppet.settings[setting]}
+        @server = Puppet.settings[setting]
+      else
+        @server = Puppet.lookup(:server) do
+          if primary_server = Puppet.settings[:server_list][0]
+            #TRANSLATORS 'server_list' is the name of a setting and should not be translated
+            Puppet.debug _("Dynamically-bound server lookup failed; using first entry from the `server_list` setting: %{server}") % {server: primary_server[0]}
+            primary_server[0]
+          else
+            setting ||= :server
+            Puppet.debug _("Dynamically-bound server lookup failed, falling back to %{setting} setting: %{server}") % {setting: setting, server: Puppet.settings[setting]}
+            Puppet.settings[setting]
+          end
         end
       end
-      server
     end
+    @server
```

Comment:
@joshcooper was this along the lines of what you were thinking about when we last talked about this bit of work? It looks like this change has caused a bunch of spec failures